hw02/src/main/java/DIYArrayList.java

```diff
@@ -0,0 +1,274 @@
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.IntFunction;
+import java.util.function.Predicate;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+public class DIYArrayList<T> implements List<T> {
+
+    private int size = 0;
+    private Object[] elements;
+
+    public <T> DIYArrayList() {
+        elements = new Object[size];
+    }
+
+    public <T> DIYArrayList(int initSize) {
+        if (initSize < 1) {
+            throw new IllegalArgumentException();
```

Comment:
Правильно, что проверяете аргументы и применили исключение.
Здесь правильнее в исключение передать больше информации, чтобы было понятно, в чем ошибка.
```suggestion
           throw new IllegalArgumentException("initSize must be > 0");
```